apps/src/codebridge/FilePreview/HTMLPreview.tsx

```diff
@@ -67,12 +100,28 @@ export const HTMLPreview = () => {
         event.origin === previewUrl
       ) {
         setCurrentFile(event.data.fileName);
+      } else if (
+        event.data.type === IframeMessageType.ADD_FILE_TO_NAVIGATION_HISTORY &&
+        event.origin === previewUrl
+      ) {
```

Comment:
I'm wondering if it's possible to simplify the navigation history logic here. `HTMLPreview` should already know the file history, because it has the `currentFile` in the url bar. Could we update our logic here such that whenever the `currentFile` changes (and we aren't loading a new page), we update the navigation history? I think we then wouldn't have to pass so many messages around navigation between the two components. From `InnerHTMLPreview`'s perspective, it would just get a file name whether the user typed in the name or clicked forward/back.